spec/controllers/internal_api/v1/pillars_controller_spec.rb

```diff
@@ -434,97 +434,107 @@
     end
   end
 
-  def expected_dex_json(num, certificate)
-    {
-      id:              num,
-      name:            "LDAP Server #{num}",
-      root_ca_data:    Base64.encode64(certificate.certificate),
-      bind:            {
-        anonymous: false,
-        dn:        "cn=admin,dc=ldap_host_#{num},dc=com",
-        pw:        "pass"
-      },
-      username_prompt: "Username",
-      user:            {
-        base_dn:  "cn=users,dc=ldap_host_#{num},dc=com",
-        filter:   "(objectClass=person)",
-        attr_map: {
-          username: "uid",
-          id:       "uid",
-          email:    "mail",
-          name:     "name"
-        }
-      },
-      group:           {
-        base_dn:  "cn=groups,dc=ldap_host_#{num},dc=com",
-        filter:   "(objectClass=group)",
-        attr_map: {
-          user:  "uid",
-          group: "member",
-          name:  "name"
-        }
-      }
-    }
-  end
-
-  # rubocop:disable RSpec/ExampleLength
-  context "with dex LDAP connectors tls" do
-    it "has dex LDAP connectors" do
-      dex_connector_ldap = create(:dex_connector_ldap, :tls, :regular_admin)
-      CertificateService.create(service: dex_connector_ldap, certificate: certificate)
+  context "with Dex LDAP connectors" do
+    before do |example|
+      # calculate a likely-unique 0-100 integer hash value by adding up the
+      # integer values of chars in the example name (context + example)
+      u = example.metadata[:full_description].codepoints.reduce(0, :+) % 100
```

Comment:
A much larger range needs to be used; supposing this number is depended on it could produce false positives during error conditions. 1-100 is not an adequate range. Also; as discussed in the other comment, if 'num' is used then this is avoided. This before block makes the code more confusing and harder to read. Please change it back and use 'num'.

To my understanding the only actual fix in this commit is changing the expect blocks to be after the get lines. These changes to the way the json is checked don't seem to be required to make the needed change.

Reply:
While 100 should be more than sufficient to handle not only the input domain here (two strings) but also the foreseeable future of "maybe a couple more" test cases, an established digest function like MD5 is somewhat less likely to collide given similar-length ASCII strings.  I didn't realize that Digest::MD5 is in the standard library, so it's easy enough to switch over.